scripts/ui/ui.ts

```diff
@@ -0,0 +1,630 @@
+import {environment} from '../../src/environments/environment';
+import {environmentTemplate} from './environment.template';
+import {Command} from 'commander';
+import fs from 'fs';
+import {EnclosureDispersalStrategy} from "../../src/app/core/testing/enums/mock-storage.enum";
+import {WebUiEnvironment} from "../../src/environments/environment.interface";
+import inquirer from 'inquirer';
+import {MockEnclosureConfig} from "../../src/app/core/testing/interfaces/mock-enclosure-utils.interface";
+import * as figlet from 'figlet';
+
+interface CommandOptions {
+  [p: string]: any;
+}
+
+interface ReportOptions {
+  data?: unknown;
+  showHeader: boolean;
+  showFooter: boolean;
+}
+
+interface Headers {
+  header: string;
+  footer: string;
+}
+
+interface ConfigWizardAnswers {
+  fileName: string;
+  controller: string;
+  shelves: string;
+  dispersal: string;
+  loadAfterSave: string;
+  saveOrCancel: string;
+}
+
+
+/*
+* Nice Header
+* */
+function banner(): string {
+  const content = figlet.textSync('TrueNAS WebUI');
+  return content;
+}
+
+function generateHeaders(content: string): Headers {
+  const width: number = 44;
+  const start = '\n ';
+  const finish = '\n';
+  const asterisk = '*';
+  const asterisksPerSide = (width - (content.length + 2)) / 2;
+  const asterisks = asterisk.repeat(asterisksPerSide);
+  const output = {
+    header: start +asterisks + ' ' + content.toUpperCase() + ' ' + asterisks + finish,
+    footer: start +asterisk.repeat(width) + '\n',
+  }
+
+  return output;
+}
+
+
+/*
+* Variables
+* */
+const environmentTs = './src/environments/environment.ts';
+const template = './scripts/ui/environment.template.ts';
+const originalEnvironment = {...environment};
+const modelsFile = './scripts/ui//models.json';
+const models: CommandOptions = JSON.parse(fs.readFileSync(modelsFile, 'utf8'));
+
+/*
+* Command Setup* */
+const program: Command = new Command()
+  .name('./ui')
+  .version('0.0.1')
+  .description('TrueNAS webui setup utility')
+  .usage('(Call from root directory)')
+  .addHelpText('before', banner())
+
+const mockExamples = `
+EXAMPLES:
```

Comment:
It has examples of how to mock different parts (model, shelves) and how to use a config file. And no examples of how to plug and play, e.g.:

```
yarn ui mock -e -m M50 -s 'es24,es24'
```

or just use existing config
```
yarn ui mock -c m50.config.json
```
That would be useful I believe.